spark3-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMerge.java

```diff
@@ -48,8 +48,77 @@ public void removeTables() {
     sql("DROP TABLE IF EXISTS source");
   }
 
-  // TODO: tests for reordering when operations succeed (both insert and update actions)
-  // TODO: tests for subqueries in conditions
+  @Test
+  public void testMergeAlignsUpdateAndInsertActions() {
+    createAndInitTable("id INT, a INT, b STRING", "{ \"id\": 1, \"a\": 2, \"b\": \"3\" }");
+    createOrReplaceView("source",
+        "{ \"id\": 1, \"c1\": -2, \"c2\": \"-3\" }\n" +
+        "{ \"id\": 2, \"c1\": -20, \"c2\": \"-30\" }");
```

Comment:
I think it would be easier to read if the strings were real strings and not numbers. If they are numbers, then I need to think about the additional concern of whether the type matches. Because I don't see a string, I don't associated each value with a particular column so I just have additional concerns in my head.

Reply:
Let me update.

Reply:
Done.